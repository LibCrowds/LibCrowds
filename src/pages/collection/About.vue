<template>

  <div id="collection-about">

    <section>
      <h2 class="text-center">About {{ collectionConfig.name }}</h2>
      <hr>
      <span
        v-if="collectionConfig.about.intro"
        v-html="collectionConfig.about.intro">
      </span>
      <p v-else>
        *{{ siteConfig.brand }}* is a platform for hosting experimental
        crowdsourcing projects from {{ siteConfig.company }}. By contributing to
        the projects on this platform you will be directly helping to enable
        future research.
      </p>
      <div class="text-center mt-3">
        <b-button
          variant="success"
          size="lg"
          :to="{
            name: 'collection-contribute',
            params: {
              collectionname: collectionConfig.key
            }
          }">
          Get Started
        </b-button>
      </div>
    </section>

    <section
      v-for="(item, index) in collectionConfig.about.subsections"
      :key="item.id"
      :id="item.id">
      <h3 class="text-center">{{ item.title }}</h3>
      <span v-html="item.markdown"></span>

      <div
        v-if="index === collectionConfig.about.subsections.length - 1"
        class="text-center">
        <hr>
        <b-button
          class="mt-1"
          variant="success"
          size="lg"
          :to="{
            name: 'collection-contribute',
            params: {
              collectionname: collectionConfig.key
            }
          }">
          Get Started
        </b-button>
      </div>

    </section>

  </div>

</template>

<script>
import siteConfig from '@/siteConfig'

export default {
  data: function () {
    return {
      siteConfig: siteConfig
    }
  },

  metaInfo () {
    return {
      title: 'About',
      description: `Learn more about the ${siteConfig.brand} platform`
    }
  },

  props: {
    collectionConfig: {
      type: Object,
      required: true
    }
  },

  computed: {
    navItems: function () {
      return this.collectionConfig.about.subsections.map((section) => {
        return { id: section.id, text: section.title }
      })
    }
  },

  mounted () {
    this.$emit('navupdated', this.navItems)
  }
}
</script>
