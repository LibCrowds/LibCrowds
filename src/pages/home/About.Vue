<template>

  <floating-tabs-layout :nav-items="navItems">

    <section id="about">
      <h2 class="text-center">About</h2>
      <hr>
      <span v-if="aboutMd" v-html="aboutMd"></span>
      <p v-else>
        {{ config.brand }} is a platform for hosting experimental crowdsourcing
        projects from {{ config.company }}. By contributing to the projects
        on this platform you will be directly helping to enable future research.
      </p>
    </section>

    <section id="technology">
      <h3>Technology</h3>
      <span v-if="technologyMd" v-html="technologyMd"></span>
      <p v-else>
        On the backend, {{ config.brand }} uses an instance of
        <a href="http://pybossa.com/" target="_blank">PyBossa</a>,
        an open-source framework for the creation of crowdsourcing projects,
        which in turn is based on the Python microframework, Flask. On the
        frontend, a custom theme has been developed using
        <a href="https://vuejs.org/" target="_blank">Vue.js</a>, a JavaScript
        framework for building user interfaces.
      </p>
    </section>

    <section id="contact">
      <h3>Contact</h3>
      <p v-if="config.forumUrl">
        Visit the <a :href="config.forumUrl">{{ config.brand }} forum</a>
        to ask any questions, report issues, or to just let us know what
        you have discovered.
      </p>
      <ul class="list-unstyled">
        <li>
          <strong>Twitter: </strong>
          <a :href="twitterUrl">
            @{{ config.contact.twitter }}
          </a>
        </li>
        <li>
          <strong>Email: </strong>
          <a :href="mailto">
            {{ config.contact.email }}
          </a>
        </li>
      </ul>

    </section>

  </floating-tabs-layout>

</template>

<script>
import config from '@/config'
import marked from 'marked'
import FloatingTabsLayout from '@/components/layouts/FloatingTabs'

export default {
  data: function () {
    return {
      config: config,
      twitterUrl: `https://twitter.com/${config.contact.twitter}`,
      mailto: `mailto:${config.contact.email}`,
      navItems: [
        { id: 'about', text: 'Top' },
        { id: 'technology', text: 'Technology' },
        { id: 'contact', text: 'Contact' }
      ]
    }
  },

  metaInfo: {
    title: 'About'
  },

  components: {
    FloatingTabsLayout
  },

  computed: {
    aboutMd: function () {
      console.log(marked(this.config.aboutMd))
      if ('aboutMd' in this.config) {
        return marked(this.config.aboutMd)
      }
      return ''
    },
    technologyMd: function () {
      if ('technologyMd' in this.config) {
        return marked(this.config.technologyMd)
      }
      return ''
    }
  }
}
</script>
