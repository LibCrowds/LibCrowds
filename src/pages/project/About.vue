<template>

  <floating-tabs-layout :nav-items="navItems">

    <section>
      <h2 class="text-center">About</h2>
      <hr>
      <span v-if="config.about.intro" v-html="config.about.intro"></span>
      <p v-else>
        {{ config.brand }} is a platform for hosting experimental crowdsourcing
        projects from {{ config.company }}. By contributing to the projects
        on this platform you will be directly helping to enable future research.
      </p>
    </section>

    <section
      v-for="(item, index) in config.about.subsections"
      :key="item.id"
      :id="item.id">
      <h3 class="text-center">{{ item.title }}</h3>
      <span v-html="item.markdown"></span>
    </section>

  </floating-tabs-layout>

</template>

<script>
import FloatingTabsLayout from '@/components/layouts/FloatingTabs'

export default {
  data: function () {
    return {
      config: require(`@/settings/projects/${this.projectname}/config`)
    }
  },

  props: {
    projectName: {
      type: String,
      required: true
    }
  },

  metaInfo: {
    title: 'About'
  },

  components: {
    FloatingTabsLayout
  },

  computed: {
    navItems: function () {
      return this.config.about.subsections.map((section) => {
        return { id: section.id, text: section.title }
      })
    }
  }
}
</script>
