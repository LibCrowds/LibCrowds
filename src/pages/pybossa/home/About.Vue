<template>
  <div id="about">
    <section class="p-3 p-md-5">
      <h1 class="text-uppercase text-center font-weight-bold display-4">
        About {{ siteConfig.title }}
      </h1>
      <hr>
      <span v-html="about"></span>
    </section>
    <section class="p-3 p-md-5">
      <div class="d-flex flex-row align-items-center justify-content-between">
        <h2 class="text-uppercase font-weight-bold display-4">Technology</h2>
        <a href="https://github.com/LibCrowds"
           class="text-muted hidden-md-down"
           target="_blank">
          <icon name="github" scale="5"></icon>
        </a>
      </div>
      <p class="mt-3">
        {{ siteConfig.title }} is part of the LibCrowds platform, which aims to
        increase access to the collections of the British Library through a
        range of experimental crowdsourcing projects. The platform was first
        launched in June 2015 with the Convert-a-Card project series, which used
        a new method for transforming printed card catalogues into electronic
        records.
      </p>
      <p>
        On the backend, each LibCrowds project series uses an instance of
        <a href="http://pybossa.com/" target="_blank">PyBossa</a>, an
        open-source framework for the creation of crowdsourcing projects, which
        in turn is based on the Python microframework, Flask. On the frontend, a
        custom theme has been developed using
        <a href="https://vuejs.org/" target="_blank">Vue.js</a>, a JavaScript
        framework for building user interfaces. A range of plugins have also
        been developed to provide additional functionality, such performing
        database searches via the Z39.50 protocol.
      </p>
      <p>
        The LibCrowds theme, all project templates and plugins are available on
        <a href="https://github.com/LibCrowds" target="_blank">GitHub</a>.
      </p>
    </section>
    <section class="p-3 p-md-5">
      <h3 class="text-uppercase font-weight-bold display-4">
        Contact {{ siteConfig.title }}
      </h3>
      <hr>
      <span v-html="about"></span>
    </section>
  </div>
</template>

<script>
import config from '@/config'
import marked from 'marked'
import 'vue-awesome/icons/github'

export default {
  name: 'about',
  data: function () {
    return {
      siteConfig: config[ this.$store.state.route.params.sitename ]
    }
  },

  computed: {
    about: function () {
      return marked(this.siteConfig.about)
    }
  }
}
</script>

<style lang="scss">
@import 'src/assets/style/_vars.scss';
@import '~bootstrap/scss/bootstrap';

section {
  &:nth-child(odd) {
    background-color: $white;
  }

  &:nth-child(even) {
    @extend .bg-faded;
  }
}
</style>
