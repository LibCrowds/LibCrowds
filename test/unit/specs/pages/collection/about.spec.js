import Vue from 'vue'
import BootstrapVue from 'bootstrap-vue'

import { shallow } from 'vue-test-utils'

import About from '@/pages/collection/About'
import testSiteConfig from 'test/testSiteConfig'
import testMicrositeConfig from 'test/testMicrositeConfig'

Vue.use(BootstrapVue)

describe('Microsite about page', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(About, {
      propsData: {
        collectionConfig: testMicrositeConfig
      }
    })
  })

  it('sets the correct data', () => {
    expect(typeof About.data).toBe('function')
    const data = About.data()
    expect(data).toEqual({
      siteConfig: testSiteConfig
    })
  })

  it('sets the correct metaInfo', () => {
    const metaInfo = About.metaInfo()
    expect(metaInfo).toEqual({
      title: 'About',
      meta: [
        {
          name: 'description',
          content: `Learn more about the ${testSiteConfig.brand} platform`
        }
      ]
    })
  })

  it('computes the correct navItems when none', () => {
    wrapper.vm.collectionConfig.about = {}
    const navItems = wrapper.vm.navItems
    expect(navItems).toEqual([])
  })

  it('computes the correct navItems when not none', () => {
    wrapper.setProps({collectionConfig: testMicrositeConfig})
    const navItems = wrapper.vm.navItems
    expect(navItems).toEqual([])
  })
})
