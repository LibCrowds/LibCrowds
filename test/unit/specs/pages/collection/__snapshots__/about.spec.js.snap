// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection about page renders correctly 1`] = `
<div id="collection-about">
  <h1 class="text-center">About undefined</h1> <span><span><h1 id="about">About</h1>
<p>Foo</p>
</span>
  <hr class="mx-0">
  </span>
  <div class="text-center">
    <a href="#" target="_self" class="btn btn-success btn-lg">
      Get Started
    </a>
  </div>
</div>
`;
